ext/rotoscope/callsite.c

```diff
@@ -0,0 +1,82 @@
+#include "ruby.h"
+#include "ruby/debug.h"
+
+#include "callsite.h"
+
+// Need the cfp field from this internal ruby structure.
+struct rb_trace_arg_struct {
+  // unused fields needed to make sure the cfp is at the
+  // correct offset
+  rb_event_flag_t unused1;
+  void *unused2;
+
+  void *cfp;
+
+  // rest of fields are unused
+};
+
+size_t ruby_control_frame_size;
+
+// MRI ruby control frames are stored in an array as of
+// ruby 2.4.1, so we depend on this to be able to get the
```

Comment:
Oops, I didn't really say that properly.  I meant to say that this is the behaviour at least until ruby 2.4.1, since I can't predict the future.